src/Symfony/Component/Security/Http/EventListener/PasswordMigratingListener.php

```diff
@@ -72,7 +82,11 @@ public function onLoginSuccess(LoginSuccessEvent $event): void
             }
         }
 
-        $passwordUpgrader->upgradePassword($user, $passwordEncoder->encodePassword($plaintextPassword, $user->getSalt()));
+        try {
+            $passwordUpgrader->upgradePassword($user, $passwordHasher->hash($plaintextPassword, $user->getSalt()));
+        } catch (InvalidPasswordException $e) {
+            throw new BadCredentialsException('Bad credentials.', 0, $e);
```

Comment:
This is already catched by `CheckCredentialsListener` (failed logins never result in password migration).

Reply:
nice catch, fixed